server.R

```diff
@@ -5,73 +5,90 @@ library(DT)
 library(reshape2)
 
 currentYear <- 2017
-currentStudentFte <- 19229
 fteYears <- c(2017:2025)
 
-exponentialGrowth <- function(years, baseFte, percentage) {
+exponentialGrowth <- function(years, initial, percentage) {
   decimalGrowth <- 1 + percentage / 100
-  yearOffsets <- years - currentYear
-  round(baseFte * decimalGrowth ^ yearOffsets)
+  round(initial * decimalGrowth ^ years)
 }
 
 shinyServer(function(input, output, session) {
 
   onRestore(function(state) {
-    x <- c(slider.args1, slider.args2, slider.args3)
+    x <- c(fte_slider, tuition_sliders, appropriation_sliders)
     lapply(seq_along(x), function(i, x){
       updateSliderInput(session, x[[i]]$inputId, value = state$values$x[[i]]$inputId)
     }, x = x)
   })
 
-  lapply(1:9, function(i, x, suffix="slider"){
-    x <- x[[i]]
-    num.lab <- x$inputId
-    slider.lab <- paste0(x$inputId, suffix)
-    observeEvent(input[[num.lab]], {
-      if(input[[num.lab]] != input[[slider.lab]])
-        onevent("mouseleave", num.lab, updateSliderInput(session, slider.lab, value = input[[num.lab]]))
-    })
-    observeEvent(input[[slider.lab]], {
-      if(input[[num.lab]] != input[[slider.lab]])
-        updateNumericInput(session, num.lab, value = input[[slider.lab]])
-    })
-  }, x = c(slider.args1, slider.args2, slider.args3))
-    
-  studentFteGrowth <- reactive({ exponentialGrowth(fteYears, currentStudentFte, input$studentFtePercentChange) })
-
-  tuitionFeesFTE <- reactive(
+  fte2017 <- 19229
+  fte2018 <- 18652
+  fte2019 <- reactive({ exponentialGrowth(1, fte2018, input$studentFtePercentChange) })
+  fte2020 <- reactive({ exponentialGrowth(1, fte2019(), input$studentFtePercentChange) })
+  fte2025 <- reactive({ exponentialGrowth(5, fte2020(), input$studentFtePercentChange) })
```

Comment:
My assumption was that whatever percentage we receive from the "through 2025" percentage slider needs to be applied individually for each year from 2021 - 2025 (inclusive). We can change this to be a single (linear) calculation instead by modifying the min/max/default values for this slider in `global.R` and changing `5` to `1` on this line. The same applies for lines 51 and 74, below.